### DISCUSSION TITLE
Net and WLan concurrent support, runtime choosing

### DISCUSSION THREAD

**Discussion by u77345** [rsta2/circle#417] [General]
Would it be possible to initialize both Net and WLan subsystems, so that Net can be tested first for the link being up? 
If Net is up, proceed with using the interface.

Otherwise, if network cable is not plugged in proceed to WLan to connect. 

Both assumed to use DHCP. Is it possible with the current state of drivers (45.3.1)?

**Comment by rsta2** [comment]
No, there can be only one network subsystem, and you have to specify, if you 
want to use Ethernet or WLAN, when it is created. Probing network interfaces 
is unfortunately not supported so far.

**Comment by hpingel** [comment]
In my project the user can decide during kernel runtime if she/he wants WLAN or ethernet to be activated on a Pi 3B+/4B. So during boot, USB stack is not necessarily initialized (for faster boot) but later on when network is needed. But there is no recovery if the user chooses the wrong thing - a reboot is necessary.

**Reply by u77345** [reply]
@hpingel It sounds interesting. How did you divide up the initialization for the two different paths?

**Reply by u77345** [reply]
Found @hpingel's implementation: https://github.com/hpingel/Sidekick64/blob/net-rebase-on-v0.51d/Source/Firmware/net.cpp
